airflow/providers/facebook/ads/hooks/ads.py

```diff
@@ -0,0 +1,152 @@
+        }
+
+    def _get_service(self) -> FacebookAdsApi:
+        """ Returns Facebook Ads Client using a service account"""
+        config = self.facebook_ads_config
+        try:
+            return FacebookAdsApi.init(app_id=config["app_id"],
+                                       app_secret=config["app_secret"],
+                                       access_token=config["access_token"],
+                                       account_id=config["account_id"],
+                                       api_version=self.api_version)
+        except AirflowException as error:
+            message = "Missing config data:\n {error}".format(error=error)
+            self._error_handler(message)
+            raise
+
+    @cached_property
+    def facebook_ads_config(self) -> None:
+        """
+        Gets Facebook ads connection from meta db and sets
+        facebook_ads_config attribute with returned config file
+        """
+        conn = self.get_connection(self.facebook_ads_conn_id)
+        if "facebook_ads_client" not in conn.extra_dejson:
+            raise AirflowException("facebook_ads_client not found in extra field")
+        return conn.extra_dejson.get("facebook_ads_client")
+
+    def bulk_facebook_report(
+        self,
+        params: Optional[Dict[str, Any]] = None,
+        fields: Optional[List[str]] = None,
+    ) -> List[AdsInsights]:
+        """
+        Pulls data from the Facebook Ads API
+
+        :param fields: List of fields that is obtained from Facebook. Found in AdsInsights.Field class.
+            https://developers.facebook.com/docs/marketing-api/insights/parameters/v6.0
+        :type fields: List[str]
+        :param params: Parameters that determine the query for Facebook
+            https://developers.facebook.com/docs/marketing-api/insights/parameters/v6.0
+
+        :return: Facebook Ads API response, converted to Facebook Ads Row objects
+        :rtype: List[AdsInsights]
+        """
+        api = self._get_service()
+        params = params if params else self.default_params
+        ad_account = AdAccount(api.get_default_account_id(), api=api)
+        _async = ad_account.get_insights(params=params, fields=fields, is_async=True)
+        try:
+            while _async.api_get()["async_status"] != JOB_COMPLETED:
+                percent = _async.api_get()["async_percent_completion"]
+                self.log.info("%s %s completed", percent, "%")
+                time.sleep(10)
+                if _async.api_get()["async_status"] in [JOB_SKIPPED, JOB_FAILED]:
+                    self._error_handler("Task failed Retry...")
+                    raise FacebookError
+            report_run_id = _async.api_get()["report_run_id"]
+            report_object = AdReportRun(report_run_id, api=api)
+            insights = report_object.get_insights()
+            return self._extract_rows(insights)
+        except FacebookRequestError as error:
+            self._error_handler(error)
```

Comment:
This will cause the errors to be repeated in the log. The first time it will be displayed by you and the second time by Airflow, because it will be an unhandled exception. I am not convinced that we need special error handling here.

Reply:
So the reason I put error handling was for formatted logs, but i think such changes done at the base level would make more sense and not in inherited hooks